TripScanner/src/main/java/com/tripscanner/TripScanner/controller/restController/ManagementItineraryRestController.java

```diff
@@ -0,0 +1,105 @@
+import org.springframework.web.bind.annotation.*;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+import java.util.Optional;
+
+import static org.springframework.web.servlet.support.ServletUriComponentsBuilder.fromCurrentRequest;
+
+@RestController
+@RequestMapping("/api/management/itineraries")
+public class ManagementItineraryRestController {
+    @Autowired
+    private ItineraryService itineraryService;
+
+    @Autowired
+    private UserService userService;
+
+    @GetMapping("")
+    public ResponseEntity<Page<Itinerary>> getItinerary(Pageable pageable) {
+        Page<Itinerary> itinerary = itineraryService.findAll(pageable);
+        if (!itinerary.isEmpty()) {
+            return ResponseEntity.ok(itinerary);
+        } else {
+            return ResponseEntity.notFound().build();
+        }
+    }
+
+
+    @PostMapping("")
+    public ResponseEntity<Itinerary> createNewItinerary(@RequestBody ItineraryDTO itinerary) throws IOException {
+
+        Optional<User> user = userService.findByUsername(itinerary.getUser());
+        if (user.isEmpty() || itinerary.getName() == null)
+            return ResponseEntity.badRequest().build();
+
+        if (itinerary.getDescription() == null)
+            itinerary.setDescription("");
+
+        Itinerary newItinerary = new Itinerary(itinerary, user.get());
+        newItinerary.setPublic(user.get().getRoles().contains("ADMIN"));
+
+        newItinerary.setImage(true);
+        Resource image = new ClassPathResource("static/img/placeholder.jpg");
+        newItinerary.setImageFile(BlobProxy.generateProxy(image.getInputStream(), image.contentLength()));
+
+        itineraryService.save(newItinerary);
+        URI location = fromCurrentRequest().path("/{id}").buildAndExpand(newItinerary.getId()).toUri();
+        return ResponseEntity.created(location).body(newItinerary);
+    }
+
+
+
+    @PutMapping("/{id}")
+    public ResponseEntity editItinerary(@PathVariable long id, @RequestBody Itinerary newItineraries) {
+        Optional<Itinerary> itinerary = itineraryService.findById(id);
+
+        if (itinerary.isPresent()) {
+            if (newItineraries.hasName()) itinerary.get().setName(newItineraries.getName());
+            if (newItineraries.hasDescription()) itinerary.get().setDescription(newItineraries.getDescription());
+            if (newItineraries.getPlaces() != null) {
+                for (Place p : newItineraries.getPlaces()) {
+                    List<Place> placeList = itinerary.get().getPlaces();
+                    placeList.add(p);
+                    itinerary.get().setPlaces(placeList);
+                }
+            }
+            itinerary.get().setPublic(newItineraries.isPublic());
+            newItineraries.setId(id);
+            itineraryService.save(itinerary.get());
+            return new ResponseEntity(HttpStatus.OK);
+        } else {
+            return ResponseEntity.notFound().build();
+        }
+    }
```

Comment:
You should use ItineraryDTO insted of an Itinerary object, change that so that it is similar to the createItinerary method.